scipy/linalg/_decomp_svd.py

```diff
@@ -133,22 +133,35 @@ def svd(a, full_matrices=True, compute_uv=True, overwrite_a=False,
     if lapack_driver not in ('gesdd', 'gesvd'):
         message = f'lapack_driver must be "gesdd" or "gesvd", not "{lapack_driver}"'
         raise ValueError(message)
-
-    if lapack_driver == 'gesdd' and compute_uv:
+    funcs = (lapack_driver, lapack_driver + '_lwork')
+    try:
+        gesXd, gesXd_lwork = get_lapack_funcs(funcs, (a1,), ilp64=True)
+        ilp64_enabled = True
+    except RuntimeError:
```

Comment:
I'm pretty sure we do not want a try-except where realistically all scipy usage goes through the except path.

Reply:
@ev-br I think it depends on whether there's an option to build SciPy with ILP64 or not (which I'm not sure is the case, but maybe you or someone can clarify). If it's possible, then I think the function mustn't crash with ValueError only because it didn't check the availability of the ILP64 functions; hence the `try-except`.

Reply:
> it depends on whether there is an option to build SciPy with ILP64 or not (which I'm not sure is the case, but maybe you or someone can clarify)

There currently isn't.

There is work towards being able to have an ILP64 build, and help is most welcome  if you're interested. It is highly unlikely to land before branching 1.15 though.

So I think it's best to fix the segfault now, and design the 64-bit behavior across the whole scipy.linalg not just one function. Not sure how it should look like, a try-except clause in every function does not look nice.

Reply:
> a try-except clause in every function does not look nice

You're right.

> There currently isn't.

In that case there is no point in checking for the availability like this. I'll simplify it.